x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/FragmentExec.java

```diff
@@ -40,25 +39,25 @@ public class FragmentExec extends LeafExec implements EstimatesRowSize {
     private final int estimatedRowSize;
 
     public FragmentExec(LogicalPlan fragment) {
-        this(fragment.source(), fragment, null, 0, null);
+        this(fragment.source(), fragment, null, 0);
     }
 
-    public FragmentExec(Source source, LogicalPlan fragment, QueryBuilder esFilter, int estimatedRowSize, PhysicalPlan reducer) {
+    public FragmentExec(Source source, LogicalPlan fragment, QueryBuilder esFilter, int estimatedRowSize) {
         super(source);
         this.fragment = fragment;
         this.esFilter = esFilter;
         this.estimatedRowSize = estimatedRowSize;
-        this.reducer = reducer;
     }
 
     private FragmentExec(StreamInput in) throws IOException {
-        this(
-            Source.readFrom((PlanStreamInput) in),
-            in.readNamedWriteable(LogicalPlan.class),
-            in.readOptionalNamedWriteable(QueryBuilder.class),
-            in.readOptionalVInt(),
-            in.getTransportVersion().onOrAfter(TransportVersions.V_8_14_0) ? in.readOptionalNamedWriteable(PhysicalPlan.class) : null
-        );
+        super(Source.readFrom((PlanStreamInput) in));
+        this.fragment = in.readNamedWriteable(LogicalPlan.class);
+        this.esFilter = in.readOptionalNamedWriteable(QueryBuilder.class);
+        this.estimatedRowSize = in.readOptionalVInt(); // TODO: fix this
+        // for old reducer
+        if (in.getTransportVersion().onOrAfter(TransportVersions.V_8_14_0)) {
```

Comment:
I am surprised that this version check here remained the same.
If I understand this right, in order to eventually remove this check here and the reducer to not be serialized at all (as this PR is doing by having each data node compute its own "reducer" and not rely on the coordinator to do it thus eliminating the "reducer" serialization completely) there should also be an EsqlCapability or version check to be used with `onOrBefore` and when the EOL is reaching that specific version we completely remove the "reducer" from the FragmentExec.

I am not saying that the current check is incorrect, but by the way it looks like now it can stay as is in this class forever, no?

Reply:
1. We don't need a new EsqlCapability, as there are no compatibility tests to verify whether the node-level reduction is executed. Since this feature is only enabled through pragmas, we can disable it in mixed clusters regardless of the pragma's value. 

2. To maintain transport BWC, we will need to send a null value for the reducer to older nodes. However, I have added a new transport version, which will allow us to completely remove the branch in V10.